app/javascript/mastodon/reducers/modal.js

```diff
@@ -11,6 +12,12 @@ export default function modal(state = initialState, action) {
     return { modalType: action.modalType, modalProps: action.modalProps };
   case MODAL_CLOSE:
     return (action.modalType === undefined || action.modalType === state.modalType) ? initialState : state;
+  case TIMELINE_DELETE:
+    if (state.modalProps.statusId === action.id) {
+      return (action.modalType === undefined || action.modalType === state.modalType) ? initialState : state;
+    } else {
+      return state;
+    }
```

Comment:
This action never has a `modalType` attribute, so you can simply do `return (state.modalProps.statusId === action.id) ? initialState : state;` here!

Reply:
Thank you. I fixed the code.